cmd/fetchd/cmd/stakereconciliationmigrate.go

```diff
@@ -0,0 +1,320 @@
+
+			genesisPath := args[0]
+			genDoc, appState, err := loadAppStateFromGenesis(genesisPath)
+			if err != nil {
+				return err
+			}
+			registrations, err := loadRegistrations()
+			if err != nil {
+				return err
+			}
+
+			bankGenesis := banktypes.GetGenesisStateFromAppState(cdc, appState)
+			authGenesis := authtypes.GetGenesisStateFromAppState(cdc, appState)
+			genesisAccounts, err := authtypes.UnpackAccounts(authGenesis.Accounts)
+			if err != nil {
+				return fmt.Errorf("unable to unpack accounts: %w", err)
+			}
+
+			stakesFile, err := os.Open(stakesCSVPath)
+			if err != nil {
+				return fmt.Errorf("unable to open staked export CSV %q: %w", stakesCSVPath, err)
+			}
+			stakesReader := csv.NewReader(stakesFile)
+
+			var (
+				ineligible         []IneligibleRegistration
+				row                []string
+				ethAddrString      string
+				oldFetchAddrString string
+				oldFetchAddr       sdk.AccAddress
+				errs               []error
+				migratedEthAddrs   = make(map[AddrString]struct{})
+			)
+			for {
+				row, err = stakesReader.Read()
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					return fmt.Errorf("failed to read stakes CSV file %s: %w", stakesCSVPath, err)
+				}
+
+				ethAddrString = normalizeEthAddr(row[0])
+				if _, ok := migratedEthAddrs[ethAddrString]; ok {
+					// Skip row if ethereum address has been successfully process previously this invocation.
+					continue
+				}
+
+				oldFetchAddrString = row[2]
+				oldFetchAddr, err = sdk.AccAddressFromBech32(oldFetchAddrString)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse bech32 address for old account %q: %w", oldFetchAddrString, err))
+					continue
+				}
+
+				registration, ok := registrations.EthAddrMap()[ethAddrString]
+				if !ok {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("no registration found for old account address %q", ethAddrString),
+					})
+					continue
+				}
+
+				newFetchAddr, err := sdk.AccAddressFromBech32(registration.NativeAddress)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse bech32 address for new account %q: %w", registration.NativeAddress, err))
+					continue
+				}
+
+				var oldAcctBalance, newAcctBalance sdk.Coins
+				var oldBalanceIndex, newBalanceIndex int
+				for i, b := range bankGenesis.Balances {
+					if b.GetAddress().Equals(oldFetchAddr) {
+						oldAcctBalance = b.GetCoins()
+						oldBalanceIndex = i
+						if newAcctBalance == nil {
+							continue
+						}
+						break
+					}
+
+					if b.GetAddress().Equals(newFetchAddr) {
+						newAcctBalance = b.GetCoins()
+						newBalanceIndex = i
+						if oldAcctBalance == nil {
+							continue
+						}
+						break
+					}
+				}
+
+				if oldAcctBalance == nil {
+					errs = append(errs, fmt.Errorf("old account with address %q does not have a balance", oldFetchAddr.String()))
+					continue
```

Comment:
not sure we need to report this, that's a noop we can ignore I guess (or add to ineligible list if we need to keep it)
```suggestion
					continue
```